app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt

```diff
@@ -0,0 +1,93 @@
+package ir.jaamebaade.jaamebaade_client.view.components
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.SheetState
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import ir.jaamebaade.jaamebaade_client.R
+import ir.jaamebaade.jaamebaade_client.viewmodel.DownloadedPoetViewModel
+import ir.jaamebaade.jaamebaade_client.wrapper.CategoryGraphNode
+
+@Composable
+@OptIn(ExperimentalMaterial3Api::class)
+fun AdvancedRandomPoemOptions(
+    modifier: Modifier = Modifier,
+    downloadedPoetViewModel: DownloadedPoetViewModel = hiltViewModel(),
+    sheetState: SheetState,
+    onDismiss: () -> Unit
+) {
+    val poetsWithCategories = downloadedPoetViewModel.categories
+    ModalBottomSheet(
+        sheetState = sheetState,
+        onDismissRequest = {
+            onDismiss()
+            downloadedPoetViewModel.saveSelectedCategoriesForRandomPoem()
+        }
+    ) {
+        Column(
+            modifier = modifier
+                .padding(10.dp)
+                .verticalScroll(rememberScrollState())
+        ) {
+            Text(
+                text = stringResource(id = R.string.ADVANCED_RANDOM_POEM_OPTIONS),
+                style = MaterialTheme.typography.titleMedium
+            )
+
+            Column {
+                poetsWithCategories?.let { categories ->
+                    RecursiveList(categories = categories)
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun RecursiveList(
```

Comment:
`RecursiveCheckList` could be a better name?

Reply:
done